cocotb/outcomes.py

```diff
@@ -44,14 +46,14 @@ def __repr__(self):
 
 
 class Error(Outcome):
-    def __init__(self, error):
-        self.error = error
+    def __init__(self, *exc_info):
+        self.exc_info = exc_info
 
     def send(self, gen):
-        return gen.throw(self.error)
+        return gen.throw(*self.exc_info)
 
     def get(self):
-        raise self.error
+        reraise(*self.exc_info)
 
     def __repr__(self):
-        return "Error({!r})".format(self.error)
+        return "Error({!r})".format(self.exc_info[1])
```

Comment:
This repr isn't sufficient to reconstruct the object any more - you probably want all three arguments